### ISSUE TITLE
Menu icon issues

### ISSUE THREAD

**Issue by 69sinisterblade69** [badspacebar/sdk#18]
When trying to set icon to boolean or slider (and possibly some other types too, not tested) I'm getting this error. But it works when I'm using it on menu type
```
menu.xerathmenu:menu("Combo", "Combo");
menu.xerathmenu.Combo:set('icon',graphics.sprite('Sprites/Combo.png')) -- works correctly

menu.xerathmenu.Combo:boolean('q_combo', 'Use Q', true);
menu.xerathmenu.Combo.q_combo:set('icon',graphics.sprite('Sprites/Combo.png')) -- doesn't work
```
![image](https://github.com/badspacebar/sdk/assets/141168740/50786f07-9035-483c-8162-356756eedd16)

**Comment by badspacebar** [comment]
It only works for main or sub menu, don't work for components.

**Comment by 69sinisterblade69** [comment]
Then it should be mentioned in documentation, or changed so it works with components.

**Comment by badspacebar** [comment]
Will be fixed in patch 20231225
